### PR TITLE
Clean up dockertools/manager.go and add more unit tests

### PR DESCRIPTION
**Pull request by yujuhong**
This function is part of the container runtime interface, so docker should
support it.

This change also adds some basic unit tests in manager_test.go. We should
start migrating docker specific tests from kubelet_test to manager_test.go.

### CODE REVIEW

**Comment by yujuhong** [comment]
/cc @vmarmol, @yifan-gu 

Not sure if `GetContainers` is needed anymore. The current use cases are:
 - kubelet.go: `GetContainerInfo`
 - container_gc.go

If we decide to remove `GetContainers` from the `Runtime` interface, I'd do that and close this PR.

**Comment by yifan-gu** [comment]
I think we had some discussion on this early, I remember @dchen1107 said the GetContainers is for getting not only kubelet containers, but also all containers so user can introspect the state of the node. So we would probably leave it in the interface IMO.

**Comment by yujuhong** [comment]
@yifan-gu, this PR assumes we are getting only the kubelet containers though. If what we want is to list all containers, I'd have to fix the PR. :)

**Comment by vmarmol** [comment]
I think we'll  remove it long-term :) From the two uses, the first we can replace with `GetPods` (I think). The second uses the Docker API directly. I think for the first version we'll disable GC in the rkt runtime as we understand more how Docker does GC. We can use `GetPods` there in the future (with an "unknown" pod having other containers not associated with a known pod).

I'm fine going this route for now though. It'll simplify what we have to move when the time comes.

File: pkg/kubelet/dockertools/manager_test.go [hunk 1]
```diff
@@ -87,3 +113,55 @@ func TestSetEntrypointAndCommand(t *testing.T) {
...
+	actualContainers, err := manager.GetContainers(false)
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+
+	actual := util.NewStringSet()
+	for _, c := range actualContainers {
+		actual.Insert(c.Name)
+	}
+	if !reflect.DeepEqual(expected.List(), actual.List()) {
```
Comment (vmarmol) [inline pkg/kubelet/dockertools/manager_test.go]: won't these be non-deterministically sorted?

**Comment by yujuhong** [comment]
@vmarmol, I agree that the use cases can probably be eliminated. I think `GetPodStatus` will be a better replacement for the first cases since we already have pod full name & uid. 

What about the reason that @yifan-gu said? Should we get all containers for users to introspect? If that's the main purpose of the function, then this PR needs to change quite a bit because most of the docker code looks at only k8s containers.

File: pkg/kubelet/dockertools/manager_test.go (hunk 1 above)
Comment (yujuhong) [inline pkg/kubelet/dockertools/manager_test.go]: `util.StringSet` actually sorts this https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/util/set.go#L122

**Comment by yifan-gu** [comment]
@yujuhong I think we can remove this from the runtime interface for now, leave a TODO and add them later...
/cc @dchen1107

**Comment by vmarmol** [comment]
LGTM.

I don't think we need it for that type of introspection, and if we do I think a better mechanism is the "unknown" pod. We can remove it from the runtime (but can we keep the rest of the refactoring you did? I think that's valuable)

**Comment by yifan-gu** [comment]
Keep the refacoring + 1

**Comment by yujuhong** [comment]
Updated the commit to remove `GetContainers` and add a unit test for `GetPods` instead.

I will remove `GetContainers` from the `Runtime` interface in another PR (since it's no longer related to this one).

**Comment by vmarmol** [comment]
LGTM, thanks @yujuhong

**Comment by yifan-gu** [comment]
LGTM

**Pull request merged**